suite/tests/api/ir_riscv64.c

```diff
@@ -1082,45 +1082,45 @@ test_jump_and_branch(void *dc)
                              opnd_create_immed_int(42, OPSZ_20b));
     pc = test_instr_encoding(dc, OP_lui, instr);
     instr = INSTR_CREATE_auipc(dc, opnd_create_reg(DR_REG_A0),
-                               opnd_create_immed_int(42, OPSZ_20b));
-    test_instr_encoding(dc, OP_auipc, instr);
+                               opnd_create_pc(pc + (3 << 12)));
+    test_instr_encoding_pc_relative(dc, OP_auipc, pc, instr);
```

Comment:
I don't see why the pc argument is related to the instruction being tested. Is it just an arbitrary pc we're using for the test? If yes, maybe test_instr_encoding_pc_relative doesn't need to take it as an arg and can generate one on its own.

Reply:
It's mainly for AUIPC, we have to pass in a pc whose difference from the current pc is aligned to `1 << 12`. But considering we're not asserting that, it should be ok to use an arbitrary value for now.

Reply:
An arbitrary value is fine. My suggestion was that test_instr_encoding_pc_relative shouldn't need to take it as an arg.

```
 If yes, maybe test_instr_encoding_pc_relative doesn't need to take it as an arg and can generate one on its own.
```

Reply:
In fact, since we're not not asserting the alignment to 1<<12 currently anymore, do we even need to use instr_encode_to_copy? Can we just use instr_encode?

Reply:
Yes, I understand that. It’s indeed cleaner without the arg. The point is once we added the assertion on AUIPC, the arg will be needed anyway, so it is no harm if we keep it as is, right?

Reply:
IMHO it's better to keep code consistent at each point. Right now, there's no need for the extra arg, so it may be confusing to a reader if we keep it.

Reply:
Sorry I missed out on this, but I just find out that the `instr_pc` arg is necessary for `ASSERT(instr_same())` at line 112 to work.

Reply:
Do you mean we need instr_pc to be set to some specific argument for that assert to work?

Reply:
Yes: 

```
    instr = INSTR_CREATE_auipc(dc, opnd_create_reg(DR_REG_A0),
                               opnd_create_pc(pc + (3 << 12)));
    test_instr_encoding_pc_relative(dc, OP_auipc, pc, instr);
```

PCs passed into `INSTR_CREATE_auipc()` and `test_instr_encoding_pc_relative()` has to align to `1<<12` so later in `test_instr_encoding_pc_relative()`, we can get the same instr back after encode and decode.

Reply:
Okay so we need it for the auipc test, but not for the others right? Maybe have some default value on the arg and   don't pass it when not needed? Also when it is not passed we can use instr_encode instead of instr_encode_to_copy.

Reply:
How about we use `test_instr_encoding_jal_or_branch()` for JAL and branch instructions, and use `test_instr_encoding_auipc()` for AUIPC. And eventually change `test_instr_encoding_jal_or_branch()` to `test_instr_encoding_jal()` once disassemble issue on branchh is fixed?

Reply:
I push said changes in https://github.com/DynamoRIO/dynamorio/pull/6208/commits/7ddef9cc7a2ead0db25aa994f9de4449046554e5.